discovery-provider/plugins/notifications/src/email/notifications/index.ts

```diff
@@ -42,11 +40,33 @@ const Results = Object.freeze({
 export const getUsersCanNotify = async (
   identityDb: Knex,
   frequency: EmailFrequency,
-  startOffset: moment.Moment
-): Promise<EmailUsers> => {
+  startOffset: moment.Moment,
+  onPage: (emailUsers: EmailUsers) => Promise<void>,
+) => {
   // const validLastEmailOffset = startOffset.subtract(2, 'hours')
-  const userRows: { blockchainUserId: number; email: string }[] =
-    await identityDb
+  let lastUser: number = Number.MAX_SAFE_INTEGER; // query uses DESC, start at max
+  const pageCount = 1000;
+  const date = new Date()
+  const time = date.getTime()
```

Comment:
nit: would prefer `Date.now()` (is faster -- see https://stackoverflow.com/questions/12517359/performance-date-now-vs-date-gettime)

Reply:
oh nice, wasn't aware of that